pulsar-testclient/src/main/java/org/apache/pulsar/proxy/socket/client/PerformanceClient.java

```diff
@@ -244,10 +242,19 @@ public void runPerformanceTest(long messages, long limit, int numOfTopic, int si
         executor.submit(() -> {
             try {
                 RateLimiter rateLimiter = RateLimiter.create(limit);
+                long startTime = System.nanoTime();
+                long testEndTime = startTime + (long) (testTime * 1e9);
                 // Send messages on all topics/producers
                 long totalSent = 0;
                 while (true) {
                     for (String topic : producersMap.keySet()) {
+                        if (testTime > 0 && System.nanoTime() > testEndTime) {
+                            log.info("------------- DONE (reached the maximum duration: [{} seconds] of production) --------------", testTime);
```

Comment:
this code will be executed by several thread concurrently.
can we set a flag and print out this like only once ?
otherwise the effect will not be veryuser-friendly

Reply:
Sorry, I understand that there is only one worker thread in the thread pool processing data (maybe we can add `--num-test-threads` to support multi-threading like `PerformanceProducer`), in this case, it will only be printed once. Am I correct?

Reply:
Can you help me look at it again? If my understanding is wrong, please explain to me and I will complete this pr as soon as possible, thx!